src/foam/nanos/dig/DUGRuleAction.js

```diff
@@ -45,68 +49,122 @@ foam.CLASS({
-                dugRule.getUrl(),
-                dugRule.evaluateBearerToken(),
-                dugDigestConfig,
-                dugRule.getFormat(),
-                new foam.lib.AndPropertyPredicate(
-                  x,
-                  new foam.lib.PropertyPredicate[] {
-                    new foam.lib.ExternalPropertyPredicate(),
-                    new foam.lib.NetworkPropertyPredicate(),
-                    new foam.lib.PermissionedPropertyPredicate()
-                  }
-                ),
-                true,
-                true
-              );
-            } else {
-              sink = new HTTPSink(
-                dugRule.getUrl(),
-                dugRule.evaluateBearerToken(),
-                dugRule.getFormat(),
-                new foam.lib.AndPropertyPredicate(
-                  x,
-                  new foam.lib.PropertyPredicate[] {
-                    new foam.lib.ExternalPropertyPredicate(),
-                    new foam.lib.NetworkPropertyPredicate(),
-                    new foam.lib.PermissionedPropertyPredicate()
-                  }
-                ),
-                true
-              );
-            }
-
-            sink.setX(x);
-            sink.put(obj, null);
-          } catch (Throwable t) {
-            ((Logger) x.get("logger")).error(this.getClass().getSimpleName(), "Error Sending DUG webhook", t);
-            var alarmDAO = (DAO) x.get("alarmDAO");
-            alarmDAO.put(
-              new Alarm.Builder(x)
-                .setName("DUG/"+rule.getDaoKey() + "/" + rule.getName())
-                .build()
+
+      if ( dugRule.getActAsUser() ) {
+        // If the actingUser is set, we want to refind the obj with that user as the subject
+        // The goal with this is to filter the permissioned properties based on acting user
+        // Instead of the user that triggered the rule
+
+        final var actingUserId = dugRule.getActingUser();
+
+        if ( actingUserId != 0 ) {
+          final var userDAO = (DAO) x.get("bareUserDAO");
+          final var user = (User) userDAO.find(actingUserId);
+
+          if ( user == null ) {
+            getLogger(x).error("DUGRule ", dugRule.getId(), " has acting user but user couldn't be found");
+            return;
+          }
+
+          final var groupDAO = (DAO) x.get("localGroupDAO");
+          final var group = groupDAO.find(user.getGroup());
+
+          if ( group == null ) {
+            getLogger(x).error("DUGRule ", dugRule.getId(), " has acting user but group couldn't be found");
+            return;
+          }
+
+          if ( SafetyUtil.isEmpty(dugRule.getSecureDaoKey()) ) {
+            getLogger(x).debug("DUGRule ", dugRule.getId(), " has acting user but secure DAO key couldn't be found");
+          }
+
+          final var daoKey = SafetyUtil.isEmpty(dugRule.getSecureDaoKey()) ? dugRule.getDaoKey() : dugRule.getSecureDaoKey();
+
+          final var actingX = x.put("group", group);
```

Comment:
actingX only overrides group but not subject. group spans spid, how do we ensure that the dug rule is only trigger for intuit?

Reply:
I'm not sure, I thoguht that we had removed the ability for spid-specific rules... @nanoscott do you know how/if we do ensure that?

Reply:
@lchanmann modified this to null out the group and set the subject, isntead of just setting the group